File: ICPC Assiut University Community/Sheet #4 (Strings)/K.cpp
#include <iostream>
#include <string>

std::string weridAddition(std::string str1, std::string str2)
{
    std::string newString;
    int lowestString = std::min(str1.length(), str2.length());

    for(int i = 0; i < lowestString; i++)
    {
        newString += str1[i];
        newString += str2[i];
    }

    if (str1.length() > str2.length())
        newString += str1.substr(lowestString);
    else 
        newString += str2.substr(lowestString);

    return newString;
}
int main()
{
    int testCases;
    std::cin >> testCases;

    while(testCases--)
    {
        std::string str1, str2;
        std::cin >> str1 >> str2;
        std::cout << weridAddition(str1, str2) <<"\n";
    }
    return 0;
}

File: ICPC Assiut University Community/Sheet #4 (Strings)/K.py
def weridAddition(string1, string2):
    newString = ""
    minLength = min(len(string1), len(string2))  


    for i in range(minLength):
        newString += string1[i]
        newString += string2[i]
    
    if len(string1) > minLength:
        newString += string1[minLength:]
    else:
        newString += string2[minLength:]
    
    return newString

testCases = int(input())

while(testCases):
    str1, str2 =  input().split()
    print(weridAddition(str1, str2))
    testCases -= 1

